### ISSUE TITLE
HAProxy leaves null bytes in the log file after log rotation

### ISSUE THREAD

**Issue by mariash** [cloudfoundry/haproxy-boshrelease#284]
When using default syslog_server value as `stdout` we are seeing that HAProxy log file `haproxy.stdout.log` keeps growing even after logs are rotated and there are null bytes at the beginning of the file.

To reproduce:

1. Generate some logs, e.g. send HTTP requests to the deployed app
2. On HAProxy VM rotate logs: `logrotate -f -v /etc/logrotate.conf`
3. See that HAProxy log is 0 KB.
4. Generate more logs
5. See that the size of HAProxy log jumped to include previous size.
6. See that the `haproxy.stdout.log` contains null bytes at the beginning of the file.

We believe that the issue is that HAProxy process keeps writing at the previous offset even though the file was truncated.

It seems like you can add a pipe when starting haproxy letting cat to handle the logging eventually instead of haproxy, e.g.

`haproxy -f ${CONFIG} -W -p ${PID_FILE} -V -db | cat –`

**Comment by maxmoehl** [comment]
Thanks for bringing this to our attention! I reproduced this issue with a minimal docker-compose setup and found out that the issue is not related to haproxy. The issue is that the redirection must happen with the `O_APPEND` flag (see also [this question](https://serverfault.com/questions/510469/null-bytes-in-a-file-created-by-2-rotated-with-logrotate)), so instead of doing `haproxy > stdout.log` we must use `haproxy >> stdout.log`. This fix works in the local testing environment I built.

Next step is to figure out where the redirection happens for our release because it's not inside our script.
